ui/desktop/electron-app/src/models/session.js

```diff
@@ -48,13 +48,18 @@ class Session {
    */
   start() {
     const command = this.cliCommand();
-    return spawnAsyncJSONPromise(command).then((spawnedSession) => {
-      this.#process = spawnedSession.childProcess;
-      this.#proxyDetails = spawnedSession.response;
-      this.#process = spawnedSession.childProcess;
-      this.#id = this.#proxyDetails.session_id;
-      return this.#proxyDetails;
-    });
+    return spawnAsyncJSONPromise(command)
+      .then((spawnedSession) => {
+        this.#process = spawnedSession.childProcess;
+        this.#proxyDetails = spawnedSession.response;
+        this.#process = spawnedSession.childProcess;
+        this.#id = this.#proxyDetails.session_id;
+        return this.#proxyDetails;
+      })
+      .finally(() => {
```

Comment:
I cleaned it up when the command finishes running instead of when the session stops because it shouldn't be needed after the command runs so doesn't need to live until the user closes the session to clean up.